### PR TITLE
make IPA download target fully customizable

### PR DESCRIPTION
**Pull request by Rozzii**
This commit:
- Makes the IPA archive's file name fully customizable
- Makes the IPA archive's file extension fully customizable

The reason for these changes is to help users who download IPA images from servers that don't follow the upstream naming conventions.

### CODE REVIEW

**Comment by Rozzii** [comment]
I had to set the DCO to pass manually, for some reason it was not detecting my signature.
/test  metal3-centos-e2e-integration-test-main

**Comment by Rozzii** [comment]
DCO was fixed properly, e-mail was set up incorrectly, thanks @tuminoid .

**Review by tuminoid — COMMENTED** [review]
/lgtm

File: get-resource.sh [hunk 1]
```diff
@@ -26,8 +26,10 @@ IPA_BASEURI="${IPA_BASEURI:-https://tarballs.opendev.org/openstack/ironic-python
 IPA_BRANCH="$(echo "${IPA_BRANCH:-master}" | tr / -)"
 IPA_FLAVOR="${IPA_FLAVOR:-centos9}"
 
-FILENAME="ipa-${IPA_FLAVOR}-${IPA_BRANCH}"
-FILENAME_EXT=".tar.gz"
+FILENAME="${FILENAME:-ipa-${IPA_FLAVOR}-${IPA_BRANCH}}"
```
Comment (dtantsur) [inline get-resource.sh:29]: I wonder if it's going to be confusing that FILENAME does not include the extension. Maybe change the code to accept a full FILENAME and strip the extension?

Comment (Rozzii) [reply get-resource.sh:29]: Okay I will change I was debating that myself too.

File: get-resource.sh [hunk 2]
```diff
@@ -39,41 +37,41 @@ TMPDIR="$(mktemp -d -p "${SHARED_DIR}"/tmp)"
...
 
     # curl didn't download anything because we have the ETag already
     # but we don't have it in the images directory
     # Its in the cache, go get it
-    ETAG="$(awk '/ETag:/ {print $2}' "${FFILENAME}.headers" | tr -d "\"\r")"
-    if [ ! -s "${FFILENAME}" ] && [ ! -e "${SHARED_DIR}/html/images/${FILENAME}-${ETAG}/${FFILENAME}" ]; then
-        mv "${SHARED_DIR}/html/images/${FFILENAME}.headers" .
-        curl_with_flags -g -O "${CACHEURL}/${FILENAME}-${ETAG}/${FFILENAME}"
+    ETAG="$(awk '/ETag:/ {print $2}' "${FILENAME}.headers" | tr -d "\"\r")"
+    if [ ! -s "${FILENAME}" ] && [ ! -e "${SHARED_DIR}/html/images/${FILENAME}-${ETAG}/${FILENAME}" ]; then
```
Comment (dtantsur) [inline get-resource.sh:56]: The part `/${FILENAME}-${ETAG}` will now work differently, it is expected? Did you mean to use `${FILENAME%%.*}`?

Comment (Rozzii) [reply get-resource.sh:56]: Yes you are right, I have missed that. I will fix it soon.

File: get-resource.sh [hunk 3]
```diff
@@ -39,41 +37,41 @@ TMPDIR="$(mktemp -d -p "${SHARED_DIR}"/tmp)"
...
     # but we don't have it in the images directory
     # Its in the cache, go get it
-    ETAG="$(awk '/ETag:/ {print $2}' "${FFILENAME}.headers" | tr -d "\"\r")"
-    if [ ! -s "${FFILENAME}" ] && [ ! -e "${SHARED_DIR}/html/images/${FILENAME}-${ETAG}/${FFILENAME}" ]; then
-        mv "${SHARED_DIR}/html/images/${FFILENAME}.headers" .
-        curl_with_flags -g -O "${CACHEURL}/${FILENAME}-${ETAG}/${FFILENAME}"
+    ETAG="$(awk '/ETag:/ {print $2}' "${FILENAME}.headers" | tr -d "\"\r")"
+    if [ ! -s "${FILENAME}" ] && [ ! -e "${SHARED_DIR}/html/images/${FILENAME%%.*}-${ETAG}/${FILENAME}" ]; then
+        mv "${SHARED_DIR}/html/images/${FILENAME}.headers" .
+        curl_with_flags -g -O "${CACHEURL}/${FILENAME}-${ETAG}/${FILENAME}"
```
Comment (dtantsur) [inline get-resource.sh:58]: The first instance of FILENAME needs the same treatment here (maybe add a FILENAME_NO_EXT variable for convenience?)

**Pull request merged**